python/cuml/metrics/pairwise_distances.pyx

```diff
@@ -136,6 +137,124 @@ def _determine_metric(metric_str, is_sparse=False):
         return PAIRWISE_DISTANCE_METRICS[metric_str]
 
 
+def nan_euclidean_distances(
```

Comment:
Function needs unit tests (or if they are already there and I missed them, just let me know)

Reply:
I have included them in the existing test cases for pairwise distances. Added nan_euclidean to pairwise_distance_metrics dictionary